editor/src/components/editor/actions/actions.tsx

```diff
@@ -4877,36 +4878,79 @@ export const UPDATE_FNS = {
     if (targetElementCoords != null && isFiniteRectangle(targetElementCoords)) {
       const isNavigatorOnTop = !editor.navigator.minimised
       const containerRootDiv = document.getElementById('canvas-root')
-      if (action.keepScrollPositionIfVisible && containerRootDiv != null) {
-        const containerDivBoundingRect = containerRootDiv.getBoundingClientRect()
-        const navigatorOffset = isNavigatorOnTop ? LeftPaneDefaultWidth : 0
+      const navigatorOffset = isNavigatorOnTop ? LeftPaneDefaultWidth : 0
+
+      // This returns the offset for 'to-origin' scroll behaviours, or used as the default
+      // for the other behaviours.
+      function canvasOffsetToOrigin(frame: CanvasRectangle): CanvasVector {
+        const baseCanvasOffset = isNavigatorOnTop ? BaseCanvasOffsetLeftPane : BaseCanvasOffset
+        return Utils.pointDifference(frame, baseCanvasOffset)
+      }
+
+      function canvasOffsetToCenter(
+        frame: CanvasRectangle,
+        bounds: CanvasRectangle | null,
+      ): CanvasVector {
+        if (bounds == null) {
+          return canvasOffsetToOrigin(frame) // fallback default
+        }
+        const xOffset = navigatorOffset > 0 ? navigatorOffset + 20 : 0
+        return canvasPoint({
```

Comment:
I think here the returned point needs to be offset with the scroll position too, now this only works correctly on elements placed with top, left 0,0 on the canvas.

https://utopia.pizza/p/a3003332-eastern-ferry/?branch_name=feat-scroll-to-center

Reply:
should be fixed with https://github.com/concrete-utopia/utopia/pull/3870/commits/a630a155d511f4c79f9378ac9d70e2737216df70